enhancements/cluster-high-availability-mode-api.md

```diff
@@ -0,0 +1,360 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+As we add more deployment topologies for OpenShift, we need a way to
+communicate expectations to the operators managing the cluster. This
+enhancement describes an API for telling operators how hard they
+should work to provide high-availability.
+
+## Motivation
+
+We want to avoid having cluster profiles use different manifests to
+configure operators managed by the `cluster-version-operator`, and
+instead rely on in-cluster APIs. This approach will make the same
+information available to operators managed by either the
+`cluster-version-operator` or `operator-lifecycle-manager`.
+
+We do not want to expose the cluster profile name directly to most of
+the operators, because we do not want to have to update operators each
+time we add a new profile.
+
+We want the setting to be global to the cluster, rather than using
+operator-specific APIs, because that simplifies the integration with
+the installer and allows it to be applied consistently in all
+operators that need them.
+
+### Goals
+
+- Expose high-availability expectations to operators running in a
+  cluster so they can configure their operands appropriately.
+
+### Non-Goals
+
+- Expose cluster profile names to operators.
+- Provide a mutable API for changing the high-availability mode (at
+  least for now).
+- Predict other capability settings that we might need to expose.
+- Define a more generic "capabilities" API.
+
+## Proposal
+
+### Cluster Operators
+
+The `infrastructure.config.openshift.io` [informational
+resource](https://docs.openshift.com/container-platform/4.6/installing/install_config/customizations.html#informational-resources_customizations),
+is expanded to include a new field to describe the high-availability
+mode desired for cluster operators running on the cluster.
+
+The `highAvailabilityMode` field will use an enum to express the
+high-availability expectations. Possible values are `Full` and
+`None`. The default is `Full`, which represents the behavior operators
+have today in a "normal" cluster. The `None` mode will be used in
+single-node deployments (developer and production), and the operator
+should not configure the operand for highly-available operation.
+
+The new cluster-wide setting will supersede existing unsupported APIs,
+such as the flags to disable high-availability requirements for
+`cluster-etcd-operator` and `cluster-authentication-operator`.
+
+If other *supported* APIs for configuring resource use exist, they
+take precedence over this new cluster-wide API. The cluster-wide API
+should be treated as a default for cases where operator-specific
+values are not set, a replacement for unsupported APIs, and a way to
+avoid adding new operator-specific APIs.
+
+### Operator Lifecycle Manager Operators
```

Comment:
This section seems at odds with https://github.com/openshift/enhancements/pull/560/files#diff-42a0d68eba341abaea58087e60f145ff6b5a01fba2fa2562c3038b831044a82bR290 where it sounds like we're saying OLM will need to set environment variables for Operators because OLM Operators may not necessarily be OpenShift specific. Which is accurate or am I misunderstanding this section?

Reply:
You're right. This section is correct, the other enhancement needs to be updated to reflect the updated approach.

Reply:
I've updated the other enhancement.